metaflow/cli.py

```diff
@@ -807,12 +752,36 @@ def start(ctx,
                                                   ctx.obj.flow,
                                                   ctx.obj.event_logger,
                                                   ctx.obj.monitor)
-    ctx.obj.datastore = DATASTORES[datastore]
+
+    ctx.obj.datastore_impl = DATASTORES[datastore]
 
     if datastore_root is None:
         datastore_root = \
-          ctx.obj.datastore.get_datastore_root_from_config(ctx.obj.echo)
-    ctx.obj.datastore_root = ctx.obj.datastore.datastore_root = datastore_root
+          ctx.obj.datastore_impl.get_datastore_root_from_config(ctx.obj.echo)
+    if datastore_root is None:
+        raise CommandException(
+            "Could not find the location of the datastore -- did you correctly set the "
+            "METAFLOW_DATASTORE_SYSROOT_%s environment variable" % datastore.upper())
+
+    ctx.obj.datastore_impl.datastore_root = datastore_root
+
+    FlowDataStore.default_backend_class = ctx.obj.datastore_impl
+    ctx.obj.flow_datastore = FlowDataStore(
+        ctx.obj.flow.name,
+        ctx.obj.environment,
+        ctx.obj.metadata,
+        ctx.obj.event_logger,
+        ctx.obj.monitor)
+
+    # It is important to initialize flow decorators early as some of the
+    # things they provide may be used by some of the objects initialize after.
+    decorators._init_flow_decorators(ctx.obj.flow,
+                                     ctx.obj.graph,
+                                     ctx.obj.environment,
+                                     ctx.obj.flow_datastore,
+                                     ctx.obj.logger,
+                                     echo,
+                                     deco_options)
```

Comment:
Looking internally (to NFLX) this call was introduced before the Metadata initialization since we wanted to register information about the project (which is a flow level decorator).

I can understand the circular dependency (newly created) since the datastore needs the metadata object - but just wanted to bring your attention to this change.

Reply:
I will check but I believe there is no use of the stuff in `current` in the intervening code but that is a good callout.

Reply:
Checked; in the internal version, the project description are not used in monitor or datastore.